packages/admin/blocks-admin/src/blocks/factories/createListBlock.tsx

```diff
@@ -308,8 +323,19 @@ export function createListBlock<T extends BlockInterface, AdditionalItemFields e
                                                     </AdminComponentStickyHeader>
                                                     <div>
                                                         {state.blocks.map((data, blockIndex) => {
-                                                            const canChangeVisibility =
-                                                                maxVisibleBlocks && totalVisibleBlocks >= maxVisibleBlocks ? data.visible : true;
+                                                            const isMinVisibleBlocksMet =
+                                                                !!minVisibleBlocks && totalVisibleBlocks >= minVisibleBlocks;
+                                                            const isMaxVisibleBlocksMet =
+                                                                !!maxVisibleBlocks && totalVisibleBlocks >= maxVisibleBlocks;
+
+                                                            const disableTooltipHoverListener =
+                                                                isMinVisibleBlocksMet &&
+                                                                totalVisibleBlocks !== minVisibleBlocks &&
+                                                                ((!isMaxVisibleBlocksMet && !data.visible) ||
+                                                                    (isMaxVisibleBlocksMet && data.visible));
```

Comment:
Could we merge the `data.visible` checks into the `isMinVisibleBlocksMet`/`isMaxVisibleBlocksMet` checks? It seems to me that these checks are always performed together.

Reply:
Could you simplify this rather complicated conditions by introducing additional variables similar to `canToggleVisibility`?

Reply:
sure

Reply:
[56beee8](https://github.com/vivid-planet/comet/pull/1865/commits/56beee8047bfeee4db156263b19e0d1a680ee26f)

If you have better variable name ideas feel free to suggest them 😂

Reply:
Just saw that the change is redundant since I removed the disableHoverListener prop